### PR TITLE
fixes for index and show pages and waypoint qr notification

### PR DESCRIPTION
**Pull request by biomoz**
Index page is only showing the user's shipment now. For employee and admin they can see every shipment.
In show (details) page users are not able to see the qr code anymore. 
On WaypointController I deleted the part for dd (debug) and duplicate code for notification because it was blocking the sending of the notification.

### CODE REVIEW

**Comment by biomoz** [comment]
Btw after the update in the web routes that added "can" functionality, you are not able to see the confirmed shipments as customer. I don't know why we need permissions to see confirmed shipments (also details) anyway but we have to fix it before demo.

**Comment by ThomasFrans** [comment]
> Btw after the update in the web routes that added "can" functionality, you are not able to see the confirmed shipments as customer. I don't know why we need permissions to see confirmed shipments (also details) anyway but we have to fix it before demo.

It's probably a mistake. You can create a small PR to fix it.

File: app/Http/Controllers/ShipmentController.php [hunk 1]
```diff
@@ -27,12 +27,24 @@ class ShipmentController extends Controller
 
     public function index(): View
     {
-        $shipments = Shipment::query()->whereNot('status', 'Awaiting Confirmation')
-            ->whereNot('status', 'Declined')
-            ->with('waypoints')
-            ->get();
-
-        return view('shipments.index', compact('shipments'));
+        if (auth()->user()->hasRole('admin') || auth()->user()->hasRole('employee')) {
```
Comment (ThomasFrans) [inline app/Http/Controllers/ShipmentController.php:30]: Never test for roles directly. There are very, very few cases where testing for a role is correct. Use permissions instead.

**Review by ThomasFrans — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/ShipmentController.php (hunk 1 above)
Comment (biomoz) [reply app/Http/Controllers/ShipmentController.php]: I have made changes and now i use view_all_shipments permission instead of roles.

**Comment by biomoz** [comment]
i got rid of the permission check when going into confirmed shipments since everyone should be able to go into that page

**Review by ThomasFrans — DISMISSED** [review]

**Review by ThomasFrans — APPROVED** [review]

**Pull request merged**